pac4j-saml/src/main/java/org/pac4j/saml/sso/impl/SAML2DefaultResponseValidator.java

```diff
@@ -164,7 +166,22 @@ protected final SAML2Credentials buildSAML2Credentials(final SAML2MessageContext
             }
         }
         return new SAML2Credentials(nameId, attributes, subjectAssertion.getConditions(),
-                SAML2Client.class.getSimpleName());
+                SAML2Client.class.getSimpleName(), sessionIndex);
+    }
+
+    /**
+     * Searches the sessionIndex in the assertion
+     * 
+     * @param subjectAssertion assertion from the response
+     * @return the sessionIndex if found in the assertion
+     */
+    private final String getSessionIndex(Assertion subjectAssertion) {
+        List<AuthnStatement> authnStatements = subjectAssertion.getAuthnStatements();
+        if(authnStatements != null && authnStatements.size() > 0) {
+        	AuthnStatement statement = authnStatements.get(0);
+        	return statement.getSessionIndex();
```

Comment:
The method `validateAuthenticationStatements` does not take into account the fact that a statement could be null (direct call to `statement.getAuthnInstant()`).

However I'm not sure about it. I'll update the PR.